### DISCUSSION TITLE
METplus Coordinated 5.1.0-beta1 Testing

### DISCUSSION THREAD

**Discussion by jprestop** [dtcenter/METplus#2039] [Release Acceptance Testing]
Listed below are summaries of the features, enhancements, and bugfixes included in the **first** development cycle for each software component of the **METplus Coordinated 5.1.0 release**. These tables summarize the status of external acceptance testing for these issues.

Please see the [METplus 5.1 Existing Builds](https://dtcenter.org/community-code/metplus/metplus-5-1-existing-builds) page to see where the beta1 releases are already installed.

<h2>External Testing Instructions</h2>

For each line in these tables, the `External Tester` listed should:
1. Test the *development release* to confirm the changes work as expected in their environment.
2. **ON THE ISSUE ITSELF**: Document their testing with a comment using the template provided below.
3. **ON THIS DISCUSSION**: Add a comment linking to their issue comment created above.

Copy/paste the following template into your **ISSUE COMMENT**:
```
<h3>Release Acceptance Testing Summary</h3>

Version: METplus-5.1.0-beta1
Date:
Location:
Status (PASS/FAIL):
Description:
```
The `External Tester` is set to `NA` for any issues that do not require external acceptance testing.

<details>
<summary><h2>METplus Wrappers <a href="https://github.com/orgs/dtcenter/projects/66/views/8">Project Board</a></h2></summary>

| Status<br>(OPEN/PASS/FAIL) | METplus Issue | External Tester | Acceptance Testing Comment Link |
|---|---|---|---|
| OPEN | dtcenter/METplus#2054 | @JiayiPeng-NOAA | |
| **PASS** | dtcenter/METplus#2049 | | No external testing required |
| **PASS** | dtcenter/METplus#2026 | | No external testing required |

</details>

<details>
<summary><h2>MET <a href="https://github.com/orgs/dtcenter/projects/64/views/2">Project Board</a></h2></summary>

<h3>List of Enhancements</h3>

| Status<br>(OPEN/PASS/FAIL)  | MET Issue | External Tester | Acceptance Testing Comment Link |
|---|---|---|---|
| **PASS** | dtcenter/MET#2146 | | No external testing required |
| **PASS** | dtcenter/MET#2386 | | No external testing required |
| **PASS** | dtcenter/MET#2388 | | No external testing required |
| **PASS** | dtcenter/MET#2415 | | No external testing required |
| **PASS** | dtcenter/MET#2421 | | No external testing required |
| OPEN | dtcenter/MET#2429 | @JiayiPeng-NOAA | |
| **PASS** | dtcenter/MET#2461 | | No external testing required |
|  **PASS** | dtcenter/MET#2469 | | No external testing required |

<h3>List of Bugfixes</h3>

See the [MET-11.0.1-rc1 Release Testing](https://github.com/dtcenter/METplus/discussions/2042) discussion for testing of these bugs for the MET-11.0.1 release.  

| Status<br>(OPEN/PASS/FAIL)  | MET Issue | External Tester | Acceptance Testing Comment Link |
|---|---|---|---|
| **PASS** | dtcenter/MET#2218 | @GwenChen-NOAA | https://github.com/dtcenter/METplus/discussions/2042#discussioncomment-5081612 |
| OPEN | dtcenter/MET#2380 | @robdarvell | |
| **PASS** | dtcenter/MET#2383 | @perthsb | Partha https://github.com/dtcenter/METplus/discussions/2042#discussioncomment-5095754, Julie https://github.com/dtcenter/MET/issues/2383#issuecomment-1440813815 |
| **PASS** | dtcenter/MET#2387 | NA | No external testing required |
| **PASS** | dtcenter/MET#2389 | NA | No external testing required |
|  **PASS** | dtcenter/MET#2390 | @HathewayWill | https://github.com/dtcenter/METplus/discussions/2042#discussioncomment-5081674 |
| **PASS** | dtcenter/MET#2399 | @malloryprow | https://github.com/dtcenter/METplus/discussions/2042#discussioncomment-5076771 |
| OPEN | dtcenter/MET#2402 | USAF? | |
| **PASS** | dtcenter/MET#2408 | NA | No external testing required |
| **PASS** | dtcenter/MET#2412 | @GwenChen-NOAA | https://github.com/dtcenter/METplus/discussions/2042#discussioncomment-5080149 |
| **PASS** | dtcenter/MET#2426 | @GwenChen-NOAA | https://github.com/dtcenter/METplus/discussions/2042#discussioncomment-5094617 |
| OPEN | dtcenter/MET#2428 | @junmeiban | |
| OPEN | dtcenter/MET#2437 | @craigr3 | |
| **PASS** | dtcenter/MET#2452 | @Ho-ChunHuang-NOAA | https://github.com/dtcenter/METplus/discussions/2042#discussioncomment-5079612 |

</details>

<details>
<summary><h2>METplus Analysis <a href="https://github.com/orgs/dtcenter/projects/68/views/15">Project Board</a></h2></summary>

| Status<br>(OPEN/PASS/FAIL)  | METdataio Issue | External Tester | Acceptance Testing Comment Link |
|---|---|---|---|
| OPEN | [dtcenter/METdataio#176](https://github.com/dtcenter/METdataio/issues/176) | | |
| OPEN | [dtcenter/METdataio#184](https://github.com/dtcenter/METdataio/issues/184) |NA|NA|
| OPEN | [dtcenter/METdataio#181](https://github.com/dtcenter/METdataio/issues/181)| ? |?|

| Status<br>(OPEN/PASS/FAIL)  | METcalcpy Issue | External Tester | Acceptance Testing Comment Link |
|---|---|---|---|
| OPEN | dtcenter/METcalcpy#270 |TBD | |
| OPEN | dtcenter/METcalcpy#279 |NA | NA|
| OPEN | dtcenter/METcalcpy#189 | TBD| |

| Status<br>(OPEN/PASS/FAIL)  | METplotpy Issue | External Tester | Acceptance Testing Comment Link |
|---|---|---|---|
| OPEN | dtcenter/METplotpy#240  |TBD |? |
| OPEN | dtcenter/METplotpy#293  |NA| NA|
| OPEN | dtcenter/METplotpy#310  |TBD |? |
| OPEN | dtcenter/METplotpy#320 |NA |NA |
| OPEN | dtcenter/METplotpy#296 |TBD |? |

| Status<br>(OPEN/PASS/FAIL)  | METviewer Issue | External Tester | Acceptance Testing Comment Link |
|---|---|---|---|
| OPEN | dtcenter/METviewer#XXXX | | |

| Status<br>(OPEN/PASS/FAIL)  | METexpress Issue | External Tester | Acceptance Testing Comment Link |
|---|---|---|---|
| OPEN | dtcenter/METexpress#XXXX | | |

</details>

**Comment by HathewayWill** [comment]
Good evening MET team,

Is enhancement from 1997 in the beta update?

https://github.com/dtcenter/METplus/discussions/1997

**Reply by jprestop** [reply]
Hi @HathewayWill. Yes, that is listed as "Bugfix: Fix issues with the met_compile_all.sh script and associated tar files ([#2390](https://github.com/dtcenter/MET/issues/2390))." in the [Release Notes](https://met.readthedocs.io/en/develop/Users_Guide/release-notes.html) under Bugfixes and was resolved with this [pull request](https://github.com/dtcenter/MET/pull/2404).

**Comment by HathewayWill** [comment]
Thank you, I was looking for #1997 didn't think to look under #2390

**Comment by AliciaBentley-NOAA** [comment]
Hi @JohnHalleyGotway @georgemccabe @jprestop. Jiayi [commented on a closed issue](https://github.com/dtcenter/MET/issues/2429) this week, so you may not have seen it. To summarize his comments:

1) I did test the consensus function by adding "write_members = FALSE;" on Hera, it works well.

2) However, in my use case for 2022 Atlantic TC verification, I set up the TC_pair configure file (see below attached).
The default setting of write_members is "TRUE", I try to modify it as "FALSE" through:
TC_PAIRS_CONSENSUS1_write_members = FALSE
TC_PAIRS_CONSENSUS2_write_members = FALSE
TC_PAIRS_CONSENSUS3_write_members = FALSE
But it does not work, The TC_pair is still doing the job to match the individual members with the best track.

**Reply by georgemccabe** [reply]
I just commented on the closed issue. The METplus config variables should be in all caps. That likely explains why the settings aren't being read into the MET config.

**Comment by JiayiPeng-NOAA** [comment]
Hi George,
Thanks for your advice. I changed the setting with
"TC_PAIRS_CONSENSUS1_WRITE_MEMBERS = FALSE".
It works now for my use-case.
The CPU difference between the "write_members = FALSE" and "write_members =
TRUE" is about 8 minutes.
It did save a few minutes by only doing the match job for ensemble mean
track and best track.
Thanks for your support!
Jiayi

**Reply by georgemccabe** [reply]
Hi @JiayiPeng-NOAA, thank you for confirming that the variable name change resolved the issue. That is great news that the setting provides a big performance boost!
